mlir/lib/Dialect/Affine/IR/AffineOps.cpp

```diff
@@ -4576,6 +4600,20 @@ AffineDelinearizeIndexOp::fold(FoldAdaptor adaptor,
   return success();
 }
 
+SmallVector<OpFoldResult> AffineDelinearizeIndexOp::getEffectiveBasis() {
+  OpBuilder builder(getContext());
+  if (hasOuterBound()) {
```

Comment:
I am trying to understand why this is not just
```
auto mixedBasis = getMixedBasis(...);
if (!hasOuterBound()) {
  return llvm::to_vector(mixedBasis.drop_front());
}
return mixedBasis;
```

Reply:
Because I was feeling clever and didn't want to pay for a reallocation I didn't need